lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java

```diff
@@ -412,94 +412,254 @@ private void standardSettings(TaskImpl task) {
+            }
+
+            // set the classification on the task object
+            task.setClassificationSummary(aClassification);
+        }
     }
 
-    private void organizeTaskAttachmentClassificationSummaries(TaskImpl taskImpl) {
-        List<Attachment> attachments = taskImpl.getAttachments();
-        List<ClassificationSummary> classificationSummaries;
-        List<String> classificationKeyList = new ArrayList<>();
-        String[] classificationKeys;
-        String domain = taskImpl.getDomain();
+    private void addWorkbasketSummariesToTaskSummaries(List<TaskSummaryImpl> taskSummaries)
+        throws NotAuthorizedException {
+        if (taskSummaries == null || taskSummaries.isEmpty()) {
+            return;
+        }
+        // calculate parameters for workbasket query: workbasket keys
+        Set<String> workbasketKeySet = new HashSet<>();
+        for (TaskSummaryImpl task : taskSummaries) {
+            workbasketKeySet.add(task.getWorkbasketSummary().getKey());
+        }
+        String[] workbasketKeyArray = workbasketKeySet.stream().toArray(String[]::new);
+        // perform workbasket query
+        List<WorkbasketSummary> workbaskets = this.workbasketService.createWorkbasketQuery()
+            .keyIn(workbasketKeyArray)
+            .list();
+        // assign query results to appropriate tasks.
+        for (TaskSummaryImpl task : taskSummaries) {
+            String workbasketKey = task.getWorkbasketSummaryImpl().getKey();
+
+            // find the appropriate classification from the query result
+            WorkbasketSummary aWorkbasket = workbaskets.stream()
+                .filter(x -> workbasketKey != null && workbasketKey.equals(x.getKey()))
+                .findFirst()
+                .orElse(null);
+            if (aWorkbasket == null) {
+                LOGGER.error("Could not find a Workbasket for task {}.", task.getTaskId());
+                throw new SystemException("Could not find a Workbasket for task " + task.getTaskId());
+            }
+            // set the classification on the task object
+            task.setWorkbasketSummary(aWorkbasket);
+        }
+    }
 
-        if (attachments != null && !attachments.isEmpty()) {
-            for (Attachment attachment : attachments) {
-                if (!classificationKeyList.contains(attachment.getClassificationKey())) {
-                    classificationKeyList.add(attachment.getClassificationKey());
+    private void addAttachmentSummariesToTaskSummaries(List<TaskSummaryImpl> taskSummaries)
+        throws NotAuthorizedException {
+        if (taskSummaries == null || taskSummaries.isEmpty()) {
+            return;
+        }
+        // calculate parameters for attachment query: task ids
+        Set<String> taskIdSet = new HashSet<>();
+        for (TaskSummaryImpl task : taskSummaries) {
+            taskIdSet.add(task.getTaskId());
```

Comment:
Möglichkeit mit 1-Zeilern (pseudo): Das wäre für alle 1-2 Zeiler for-loops. Optionale Anmerkung.

List<String> idList = taskSummaries.stream().map(t -> t.getTaskId()).collect(Collectors.toList());
taskIdSet.addAll(idList);